src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1195,6 +1202,18 @@
        (remove (comp :query-only? val))
        keys))
 
+(defn projectable-json-fields
+  "Returns a list of fields of queryable json fields whose
+   field type supports json subtree extraction.
+
+   Currently only queryable json fields whose :field value
+   is a keyword or raw SQL support json subtree extraction."
+  [{:keys [projections]}]
+  (->> projections
+       (filter (comp #(and (= :queryable-json (:type %))
+                           (:projectable-json? %)) val))
```

Comment:
Think val should be on a separate line so it is more obviously a comp argument than an additional and clause.